catalog/imaging/catalog-imaging-transformer/src/main/java/com/connexta/alliance/transformer/nitf/NitfPreStoragePlugin.java

```diff
@@ -91,44 +94,72 @@ public UpdateStorageRequest process(UpdateStorageRequest updateStorageRequest)
+        }
+        return false;
+    }
+
     private void process(List<ContentItem> contentItems) {
-        ContentItem contentItem = contentItems.get(0);
+        List<ContentItem> newContentItems = new LinkedList<>();
+        contentItems.forEach(contentItem -> process(contentItem).ifPresent(newContentItems::add));
+        contentItems.addAll(newContentItems);
+    }
+
+    private Optional<ContentItem> process(ContentItem contentItem) {
         Metacard metacard = contentItem.getMetacard();
 
+        if (!isNitfMimeType(contentItem.getMimeTypeRawData())) {
+            LOGGER.debug("skipping content item: filename={} mimeType={}",
+                    contentItem.getFilename(),
+                    contentItem.getMimeTypeRawData());
+            return Optional.empty();
+        }
+
         try {
             BufferedImage renderedImage = renderImage(contentItem);
 
             if (renderedImage != null) {
                 addThumbnailToMetacard(metacard, renderedImage);
-                ContentItem overviewContentItem = createOverview(contentItem.getId(), renderedImage,
+                ContentItem overviewContentItem = createOverview(contentItem.getId(),
+                        renderedImage,
                         metacard);
 
                 if (overviewContentItem != null) {
-                    contentItems.add(overviewContentItem);
+                    return Optional.of(overviewContentItem);
                 }
             }
         } catch (IOException | ParseException e) {
             LOGGER.warn(e.getMessage(), e);
         }
+
+        return Optional.empty();
     }
 
     private BufferedImage renderImage(ContentItem contentItem) throws IOException, ParseException {
         NitfReader reader = new NitfInputStreamReader(contentItem.getInputStream());
         AllDataExtractionParseStrategy parsingStrategy = new AllDataExtractionParseStrategy();
         NitfFileParser.parse(reader, parsingStrategy);
 
-        if (parsingStrategy.getImageSegmentData().size() == 0) {
+        if (parsingStrategy.getImageSegmentData()
+                .size() == 0) {
             return null;
         }
 
-        NitfImageSegmentHeader header = parsingStrategy.getImageSegmentHeaders().get(0);
-        byte[] image = parsingStrategy.getImageSegmentData().get(0);
+        NitfImageSegmentHeader header = parsingStrategy.getImageSegmentHeaders()
+                .get(0);
+        byte[] image = parsingStrategy.getImageSegmentData()
+                .get(0);
 
         NitfRenderer renderer = new NitfRenderer();
-        ImageInputStream inputStream = new MemoryCacheImageInputStream(
-                new ByteArrayInputStream(image));
-        BufferedImage bufferedImage = renderer.render(header, inputStream);
-        return bufferedImage;
+        ImageInputStream inputStream = new MemoryCacheImageInputStream(new ByteArrayInputStream(
+                image));
+        return renderer.render(header, inputStream);
```

Comment:
I prefer the assignment because when debugging you can see the final value before it's returned while still being in the stackframe.